File: src/java/BubbleSort.java
public class BubbleSort {

  public static void main(String[] args) {

    int vector[] = {9, 0, 4, 2, 3, 8, 7, 1, 6, 5};

    System.out.println("Bubble Sort:");
    System.out.println("Unordered vector:");
    showVector(vector);

    vector = bubbleSort(vector);

    System.out.println("Sorted vector:");
    showVector(vector);
  }

  public static int[] bubbleSort(int vector[]) {
    int swap = 0;
    for (int i = 0; i < vector.length - 1; i++) {
      if (vector[i] > vector[i + 1]) {
        int temp = vector[i];
        vector[i] = vector[i + 1];
        vector[i + 1] = temp;
        swap++;
      }
    }
    if (swap != 0) bubbleSort(vector);
    return vector;
  }

  public static int[] bubbleSort2(int vector[]) {
    for (int i = 1; i < vector.length; i++) {
      int length = vector.length;
      for (int j = 0; j < length - i; j++) {
        if (vector[j] > vector[j + 1]) {
          int temp = vector[j];
          vector[j] = vector[j + 1];
          vector[j + 1] = temp;
        }
      }
    }
    return vector;
  }

  public static void showVector(int vector[]) {
    for (int i = 0; i < vector.length; i++) {
      System.out.print(vector[i] + ", ");
    }
    System.out.println("");
  }
}

File: src/python/bubble_sort.py
""" Recursive Bubble Sort in Python """


def bubble_sort(data, size):
    """
    Implementation of a bubble sort algorithm with recursion.

    Arguments:
    data: list. List to be sorted
    size: int. List size

    Returns the ordered "data" list.
    """
    swap = False
    for i in range(0, size - 1):
        if data[i] > data[i + 1]:
            data[i], data[i + 1] = data[i + 1], data[i]
            swap = True
    if swap:
        bubble_sort(data, size - 1)


if __name__ == "__main__":
    data = [2, 9, 8, 0, 1, 3, 5, 4, 6, 7]
    print(data)
    bubble_sort(data, len(data))
    print(data)
